### DISCUSSION TITLE
Session Replay for Mobile is Now in Open Beta

### DISCUSSION THREAD

**Discussion by bruno-garcia** [getsentry/sentry#74322] [Launch Feedback]
# EDIT: Session Replay for Mobile is Generally available

-------- 

### Original Beta post:

We’re excited to finally announce that **Session Replay is now available for mobile applications** in open beta for React Native, iOS and Android.

Session Replay lets developers see a visual reproduction of user engagement with their app, so you can understand **when, where, and how an error is impacting your app without having to repro it yourself or talk to a customer**. With Replay, you have full visibility into how issues such as crashes, application hangs, and high latency occur. You can also assess how severe an issue is depending on the impact on the UI or how the user responded.

Replay for mobile includes details **like battery life, OS version, device specs, network requests, and exceptions** that help you understand what’s happening going on with your app behind the scenes to get to the root of the problem.

![image](https://github.com/user-attachments/assets/c3ce5350-6075-498d-9e59-f0c29009a967)

## **Get Started**

You can install mobile replay by just updating your SDK and a few lines of code. See the set up instructions for Android, iOS, and React Native SDKs below. Support for [Flutter](https://docs.sentry.io/platforms/flutter/) is being tracked on this [GitHub issue](https://github.com/getsentry/sentry-dart/issues/1193).

* [iOS](https://docs.sentry.io/platforms/apple/guides/ios/session-replay/)
* [Android](https://docs.sentry.io/platforms/android/session-replay/)
* [React Native](https://docs.sentry.io/platforms/react-native/session-replay/)

Sentry is privacy-first; as such, our privacy configuration by default is that all text and images are redacted to ensure no sensitive user information is collected. However, if you don’t have any sensitive information on your website, you can opt-out of the redaction (see docs above for more information). For product documentation and a FAQ, check out the [product docs](https://docs.sentry.io//product/explore/session-replay/mobile/).

## We Want to Hear Your Feedback

If you have any questions or concerns, please let us know. Once you’ve tried mobile replay, please share your feedback below. 👇 If you would like to report a bug, please open a GitHub issue with a link to a relevant replay ([Android](https://github.com/getsentry/sentry-java/issues/new/choose), [iOS](https://github.com/getsentry/sentry-cocoa/issues/new/choose) and [React Native](https://github.com/getsentry/sentry-react-native/issues/new/choose)).

**Comment by tday** [comment]
Congrats! The session replay URL for iOS links to React Native

**Reply by kahest** [reply]
Thanks! I updated the link :)

**Comment by jas-kas** [comment]
FYI We're hosting a workshop next week Wednesday on how to solve mobile bugs with Session Replay. This is your chance to ask your questions on Mobile Replay directly to our eng team and see a demo of Mobile Replay live. 🎉 You can sign up here: https://sentry.io/resources/solve-mobile-bugs-with-session-replay/

**Comment by kitTiwari** [comment]
Is sentry replay available for self hosted/on-premise latest version ?

**Reply by bruno-garcia** [reply]
During the beta we're not actively targeting self hosted. That said, there's nothing we did to make it 'not work' on SH. If you manage to give it a try, please reply here if it works as I'm sure it'll be useful to others.
Once we get close to GA we'll make sure things work properly on SH too!

**Reply by garudamon** [reply]
the replay for react native is not working with SH, I have compared it
<img width="487" alt="sentry-cloud-service" src="https://github.com/user-attachments/assets/87ce4f21-5c93-4280-83df-1ddc93f6a14d">
<img width="707" alt="sentry-self-hosted" src="https://github.com/user-attachments/assets/44fa3a97-cd05-44ce-bf0e-57ce5664a752">

**Reply by garudamon** [reply]
@jas-kas hi jasmine, do you have any schedule for releasing the support of replay on self hosted?

**Reply by jas-kas** [reply]
Hey @garudamon! Thanks for reaching out. Unfortunately, this will not be until another few months as we're prioritizing redaction capabilities, improving performance overhead impact, more sampling options, etc.

**Reply by Eyesonly88** [reply]
Hi @jas-kas , it is now August 29, 2025. I don't see any mention of this on the self-hosted releases on Gthub. Can you please confirm if this is now available in the self-hosted version of Sentry or not?
If it is available, which version? Thank you 🙏

**Reply by jas-kas** [reply]
Hey @Eyesonly88! Mobile Session Replay is available for self-hosted and will be available on any recent release (approx v24.7.1 or later https://github.com/getsentry/sentry/releases?page=3). If you have any questions on getting this set up, or any challenges, please open an issue in our `self-hosted` repo: https://github.com/getsentry/self-hosted/issues. Thank you~

**Comment by jtavio** [comment]
Hello good afternoon, does this also work for flutter?

**Reply by jas-kas** [reply]
Hey @jtavio!

Yes, we now support Flutter. Please see [v8.9.0](https://github.com/getsentry/sentry-dart/releases/tag/8.9.0) of our Flutter SDK for instructions.

**Comment by Legion2** [comment]
How can I add custom tags to session replays so I can filter replays in sentry by the custom tags? Tags I set with `beforeSend` do not apply to the session replays.

**Reply by jas-kas** [reply]
Hey @Legion2 👋 

To help answer this, what platform do you use today? Is it RN, iOS, Android or something else?

**Reply by Legion2** [reply]
RN, android, iOS and Kotlin multiplatform

**Reply by romtsn** [reply]
@Legion2 which SDK do you want to call `beforeSend` from? In general, `Sentry.setTag()` should work for replays already, but beforeSend might not be available for all platforms

**Reply by Legion2** [reply]
currently we call beforeSend in RN, Android and ios, because in RN we handle js error and on android and ios we handle native exceptions. We already call `event.setTag()` in beforeSend. Should we use Sentry.setTag() instead?

**Reply by romtsn** [reply]
yeah, so the replay events are only available on iOS through `beforeSend`. We'll add something similar on Android, but I'm not sure if we can make it work for RN, because it'd require syncing the entire event over the bridge, which is expensive (the replay runs entirely on the native SDKs, RN just wraps some options and customization). So for the time being `Sentry.setTag` is the most reliable way to set a tag for replay events I believe (calling this in RN should work too).

**Reply by Legion2** [reply]
thats fine, we only use beforeSend in js and native, because js events are not processed in native, so we can only process them in js. If replays are only available in native beforeSend that is fine.

**Comment by Legion2** [comment]
Is there an equivalent mobile API for the javascript `Replay.start/stop/flush` apis? https://docs.sentry.io/platforms/javascript/session-replay/understanding-sessions/#manually-flushing-recording-data

**Reply by jas-kas** [reply]
Hey @Legion2, there is not an equivalent mobile API just yet. We're working on a similar request to pause/resume recordings here: https://github.com/getsentry/sentry/issues/63248

**Reply by Legion2** [reply]
I need the flush API in particular, to allow users to manually report problems in the UI

**Reply by jas-kas** [reply]
Gotcha! Follow up ticket for flush API: https://github.com/getsentry/sentry/issues/78018

**Comment by Legion2** [comment]
I use RN session replays and configured session replay with `errorSampleRate = 1.0` but don't get replays when the error is captured in js. If the error is captured in native it will create a replay.

**Reply by romtsn** [reply]
how do you capture an error exactly? I believe the event should contain exceptions to be qualified as error: https://github.com/getsentry/sentry-react-native/blob/fd21b24ddb99bff181a50c4fca9f0c88846cf022/src/js/replay/mobilereplay.ts#L81-L85

**Reply by Legion2** [reply]
thanks for the reference I will check that

**Reply by Legion2** [reply]
OK found my error, I removed `Sentry.mobileReplayIntegration()` from js because I moved all the initialization of the replay stuff to native, but `Sentry.mobileReplayIntegration()` is still needed in js to trigger the replay on js errors.

**Reply by Legion2** [reply]
I now added it back to js and it sometimes triggers a replay for an js error.

**Comment by felipenwelter** [comment]
Do you have any plans on the timeline for releasing an official version?

**Reply by bruno-garcia** [reply]
The current plan is to GA sometime in November. We'll share the official date soon.

**Reply by jas-kas** [reply]
^ Update to this, we're looking at Jan 2025.

**Comment by Legion2** [comment]
```
Thread Performance Checker: Thread running at User-initiated quality-of-service class waiting on a lower QoS thread running at Default quality-of-service class. Investigate ways to avoid priority inversions
0   App                                 0x0000000106c6ca48 $s6Sentry0A14OnDemandReplayC11renderVideo33_70FE3B80E922CEF5576FF378226AFAE1LL4with4from2atAA0aF4InfoCSgSayAA0aD5FrameVG_Siz10Foundation3URLVtKF + 2132
1   App                                 0x0000000106c6be34 $s6Sentry0A14OnDemandReplayC15createVideoWith9beginning3endSayAA0aF4InfoCG10Foundation4DateV_ALtKF + 1284
2   App                                 0x0000000106c6c0c0 $s6Sentry0A14OnDemandReplayC15createVideoWith9beginning3endSayAA0aF4InfoCG10Foundation4DateV_ALtKFTo + 228
3   App                                 0x0000000106c86984 $s6Sentry0A13SessionReplayC16createAndCapture33_0C34F63BF2CAC7EBBCDDA9F08A50850ALL9startedAt10replayTypey10Foundation4DateV_AA0acQ0OtFyycfU_ + 500
4   App                                 0x0000000106c48e84 $sIeg_IeyB_TR + 48
5   App                                 0x0000000106ba3268 __53-[SentryDispatchQueueWrapper dispatchAsyncWithBlock:]_block_invoke + 56
6   libdispatch.dylib                   0x00000001101bca30 _dispatch_call_block_and_release + 32
7   libdispatch.dylib                   0x00000001101be71c _dispatch_client_callout + 20
8   libdispatch.dylib                   0x00000001101c65e8 _dispatch_lane_serial_drain + 828
9   libdispatch.dylib                   0x00000001101c7360 _dispatch_lane_invoke + 408
10  libdispatch.dylib                   0x00000001101d45f0 _dispatch_root_queue_drain_deferred_wlh + 328
11  libdispatch.dylib                   0x00000001101d3c00 _dispatch_workloop_worker_thread + 580
12  libsystem_pthread.dylib             0x00000002280bfc7c _pthread_wqthread + 288
13  libsystem_pthread.dylib             0x00000002280bc488 start_wqthread + 8
```
I get the following warning in xcode when using session replay on ios.

**Reply by bruno-garcia** [reply]
@brustolin have we seen this one?

**Reply by brustolin** [reply]
I will take a look

**Reply by brustolin** [reply]
This will be fixed in the next version

**Comment by jas-kas** [comment]
Mobile Session Replay has been generally available since the beginning of this year, hence I'll be closing this discussion. If you have any questions or encounter problems, please open a GitHub issue in the relevant platform repository (e.g., [sentry-cocoa](https://github.com/getsentry/sentry-cocoa/issues) for iOS or [sentry-react-native](https://github.com/getsentry/sentry-react-native/issues) for React Native)
